### ISSUE TITLE
Bad f-string, causing CI issues

### ISSUE THREAD

**Issue by fanninpm** [RustPython/RustPython#3531]
#### Code
```python
foo = "bar"
part_spec = ">+#10x"
print(f"{16:0{part_spec}}{foo}")
```

#### CPython
```text
00000+0x10bar
```

#### RustPython
```text
SyntaxError: Got error in f-string: Invalid expression: invalid syntax. Got unexpected token 'part_spec' at line 1 column 9
print(f"{16:0{part_spec}}{foo}")
        ^
```

#### Code
```python
import decimal
width = 10
precision = 4
value = decimal.Decimal('12.34567')
print(f'result: {value:{width}.{precision}}')
```

#### CPython
```text
result:      12.35
```

#### RustPython
```text
SyntaxError: Got error in f-string: Invalid expression: invalid syntax. Got unexpected token '{' at line 1 column 9
print(f'result: {value:{width}.{precision}}')
        ^
```

## Python Documentation

https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-python-grammar-format_spec

@coolreader18 this has been plaguing the CI for the past day or so (because of the `--install-pip` step).

**Comment by gavindsouza** [comment]
I just tried `rustpython get-pip.py` on latest HEAD (bd295e0f44870d534565ffa7cb1be9c93b11c7e1) and it threw

```bash
 File "/Users/Shared/projects/RustPython/vm/pylib-crate/Lib/zipimport.py", line 191, in get_filename
    code, ispackage, modpath = _get_module_code(self, fullname)
  File "/Users/Shared/projects/RustPython/vm/pylib-crate/Lib/zipimport.py", line 713, in _get_module_code
    code = _compile_source(modpath, data)
  File "/Users/Shared/projects/RustPython/vm/pylib-crate/Lib/zipimport.py", line 647, in _compile_source
    return compile(source, pathname, 'exec', dont_inherit=True)
SyntaxError: Got error in f-string: Invalid expression: invalid syntax. Got unexpected token ',' at line 408 column 27
        completed_str = f"{completed_ratio:,.{precision}f}"
                          ^
```
---

Extracted example:

On CPython 3.9.9:

```python
>>> completed_ratio = 1212.121212
>>> precision = 2
>>> f"{completed_ratio:,.{precision}f}"
'1,212.12'
```

On Rust Python 0.1.2:

```python
>>>>> completed_ratio = 1212.121212
>>>>> precision = 2
>>>>> f"{completed_ratio:,.{precision}f}"
SyntaxError: Got error in f-string: Invalid expression: invalid syntax. Got unexpected token ',' at line 1 column 3
f"{completed_ratio:,.{precision}f}"
  ^
```
---

@fanninpm Does your PR fix this too?

**Comment by youknowone** [comment]
Once this is fixed, #3535 need to be reverted
